lib/puppet/provider/exec/powershell.rb

```diff
@@ -68,7 +68,8 @@ def run(command, check = false)
         return super("cmd.exe /c \"\"#{native_path(command(:powershell))}\" #{legacy_args} -Command - < \"#{native_path}\"\"", check)
       end
     else
-      result = ps_manager.execute(command)
+      timeout_ms = resource[:timeout].nil? ? nil : resource[:timeout].to_i * 1000
```

Comment:
The exec type casts as a float so i'm not sure `to_i` is needed? (https://github.com/puppetlabs/puppet/blob/master/lib/puppet/type/exec.rb#L283-L290) We may lose some precision actually if we do.